# Star Dataset Analysis

In [1]:
import pandas as pd
import numpy as np

## Research Question

Is there a relationship between a star's radius and its temperature?

## Step 1
##### Load Dataset

In [4]:
df = pd.read_csv("../data/Stars.csv.xls")

## Step 2

#### Calculate the correlation between radius and temperature

In [6]:
df[["Temperature", "R"]].head()
df[["Temperature", "R"]].describe()
df[["Temperature", "R"]].info()
df[["Temperature", "R"]].isnull().sum()

<class 'pandas.DataFrame'>
RangeIndex: 240 entries, 0 to 239
Data columns (total 2 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Temperature  240 non-null    int64  
 1   R            240 non-null    float64
dtypes: float64(1), int64(1)
memory usage: 3.9 KB


Temperature    0
R              0
dtype: int64

## Step 3

#### Calculate the correlation between radius and temperature

In [8]:
df[["Temperature", "R"]].corr()

,Temperature,R
Temperature,1.000000,0.064216
R,0.064216,1.000000


## Conclusion

The correlation coefficient between radius and temperature is 0.064, indicating a very weak positive relationship. Based on this dataset, there is no strong linear correlation between a star's radius and its temperature.